x-pack/legacy/plugins/ml/public/data_frame/pages/data_frame_new_pivot/components/step_define/step_define_form.tsx

```diff
@@ -450,31 +529,131 @@ export const StepDefineForm: SFC<Props> = React.memo(({ overrides = {}, onChange
                 <span>{kibanaContext.currentIndexPattern.title}</span>
               </EuiFormRow>
               {!disabledQuery && (
-                <EuiFormRow
-                  label={i18n.translate('xpack.ml.dataframe.stepDefineForm.queryLabel', {
-                    defaultMessage: 'Query',
-                  })}
-                  helpText={i18n.translate('xpack.ml.dataframe.stepDefineForm.queryHelpText', {
-                    defaultMessage: 'Use a query to filter the source data (optional).',
-                  })}
-                >
-                  <KqlFilterBar
-                    indexPattern={indexPattern}
-                    onSubmit={searchHandler}
-                    initialValue={searchString === defaultSearch ? emptySearch : searchString}
-                    placeholder={i18n.translate(
-                      'xpack.ml.dataframe.stepDefineForm.queryPlaceholder',
+                <Fragment>
+                  {!isAdvancedSourceEditorEnabled && (
+                    <EuiFormRow
+                      label={i18n.translate('xpack.ml.dataframe.stepDefineForm.queryLabel', {
+                        defaultMessage: 'Query',
+                      })}
+                      helpText={i18n.translate('xpack.ml.dataframe.stepDefineForm.queryHelpText', {
+                        defaultMessage: 'Use a query to filter the source data (optional).',
+                      })}
+                    >
+                      <KqlFilterBar
+                        indexPattern={indexPattern}
+                        onSubmit={searchHandler}
+                        initialValue={searchString === defaultSearch ? emptySearch : searchString}
+                        placeholder={i18n.translate(
+                          'xpack.ml.dataframe.stepDefineForm.queryPlaceholder',
+                          {
+                            defaultMessage: 'e.g. {example}',
+                            values: { example: 'method : "GET" or status : "404"' },
+                          }
+                        )}
+                      />
+                    </EuiFormRow>
+                  )}
+                </Fragment>
+              )}
+            </Fragment>
+          )}
+
+          {isAdvancedSourceEditorEnabled && (
+            <Fragment>
+              <EuiFormRow
+                label={i18n.translate(
+                  'xpack.ml.dataframe.stepDefineForm.advancedSourceEditorLabel',
+                  {
+                    defaultMessage: 'Source configuration object',
```

Comment:
As above, I think `Source query clause` might be a more accurate description here.

Reply:
261ed3bc5b2a487226934e5f31aa2f5c9c3ebf1f